src/main/java/nextstep/subway/auth/application/AuthenticationResolver.java

```diff
@@ -0,0 +1,33 @@
+package nextstep.subway.auth.application;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import nextstep.subway.auth.domain.Authentication;
+import nextstep.subway.auth.domain.AuthenticationPrincipal;
+import nextstep.subway.auth.infrastructure.SecurityContext;
+import nextstep.subway.auth.infrastructure.SecurityContextHolder;
+
+@Component
+public class AuthenticationResolver implements HandlerMethodArgumentResolver {
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        return parameter.getParameterType().equals(AuthenticationPrincipal.class);
```

Comment:
우선 이 부분이 문제인 것 같습니다.
supportParameter에서 false 값이 응답되어 아래의 resolveArgument를 호출하지 못하는 것 같네요.

이 구문을 아래와 같이 변경해보면 어떨까요?
```suggestion
        return parameter.hasParameterAnnotation(AuthenticationPrincipal.class);
```

Reply:
@boorownie 찾아보니 ```getParameterType()```은 
`
Return the type of the method/constructor parameter.
`
라는 설명이 있네요.
그와 달리 ```hasParameterAnnotation()```은
`
Return true if the parameter has at least one annotation, false if it has none.
`
라고 하는 군요. 
그래서 커스텀 어노테이션은 ```getParameterType()```에서는 판별 기능이 작동하지 않았다고 생각하면 될까요?

Reply:
네 정확히 맞습니다 :)